### PR TITLE
IGNITE-11380 Add support of JAR files to UriDeploymentSpi

### PR DESCRIPTION
**Pull request by dmekhanikov**

### CODE REVIEW

File: modules/extdata/uri/pom.xml [hunk 1]
```diff
@@ -128,6 +220,32 @@
...
+                            <target>
+                                <mkdir dir="${basedir}/target/file_tmp/classes/org/apache/ignite/spi/deployment/uri/tasks/" />
+                                <copy file="${basedir}/target/classes/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestTask11.class" todir="${basedir}/target/file_tmp/classes/org/apache/ignite/spi/deployment/uri/tasks/" />
+                                <copy file="${basedir}/target/classes/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestTask11.class" tofile="${basedir}/target/file_tmp/classes/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestWithNameTask11.class" />
+
+                                <jar destfile="${basedir}/target/file/bad-signed-deployfile.jar" basedir="${basedir}/target/file_tmp/classes" />
+
+                                <signjar jar="${basedir}/target/file/bad-signed-deployfile.jar" keystore="${basedir}/config/signeddeploy/keystore" storepass="abc123" keypass="abc123" alias="business" />
+
+                                <sleep seconds="2" />
```
Comment (alamar) [inline modules/extdata/uri/pom.xml]: Why?

File: modules/extdata/uri/src/main/java/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestWithNameTask10.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.ignite.compute.ComputeTaskSplitAdapter;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * URI deployment test task with name.
+ */
+@ComputeTaskName("GridUriDeploymentTestWithNameTask10")
+public class GridUriDeploymentTestWithNameTask10 extends ComputeTaskSplitAdapter<Object, Object> {
```
Comment (alamar) [inline modules/extdata/uri/src/main/java/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestWithNameTask10.java]: What's the point of these three empty tasks? Maybe log something at least?

File: modules/urideploy/src/main/java/org/apache/ignite/spi/deployment/uri/GridUriDeploymentUnitDescriptor.java [hunk 3]
```diff
@@ -142,17 +143,17 @@ public Type getType() {
...
+     * @return {@code null} if tasks were deployed directly and reference to the package file URI if tasks
+     * were deployed from the file.
      */
     public String getUri() {
         return uri;
     }
 
     /**
-     * Tasks GAR file.
+     * Tasks package file.
```
Comment (alamar) [inline modules/urideploy/src/main/java/org/apache/ignite/spi/deployment/uri/GridUriDeploymentUnitDescriptor.java]: Task's? Tasks'?

File: modules/extdata/uri/src/main/java/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestWithNameTask10.java (hunk 2 above)
Comment (dmekhanikov) [reply modules/extdata/uri/src/main/java/org/apache/ignite/spi/deployment/uri/tasks/GridUriDeploymentTestWithNameTask10.java]: We don't even run them. We only check if we can load these classes. So, it doesn't make much sense to put anything inside.

File: modules/extdata/uri/pom.xml (hunk 1 above)
Comment (dmekhanikov) [reply modules/extdata/uri/pom.xml]: If JAR assembly and running this Ant script happens within one second, then the `zip` task won't detect any changes and leave the JAR not updated.
This `sleep` and `touch` bellow update the timestamp of the file, so the `zip` task doesn't ignore the file.
The same thing happens in the code bellow, where a similar GAR file is assembled.

**Comment by dmekhanikov** [comment]
@alamar This behaviour is changed actually. If you have JARs lying around your repository, then they will be also loaded. 
We could introduce a new configuration property to make people enable this new feature and acknowledge the fact, that their JARs will also be loaded now. But I don't think, it's too much of a problem, so another piece of configuration seems to be an overkill to me.

**Pull request closed without merging**